### ISSUE TITLE
[Feature Request] Bulk selection (mainly in list-view)

### ISSUE THREAD

**Issue by kaibs** [crocodilestick/Calibre-Web-Automated#137]
First, I want to thank you for this great project. It really is a very nice improvement over the “stock” calibre-web!

Is your feature request related to a problem? Please describe.

- I've been playing around with it for a few days now, and there is one feature I've been really missing. When using the “Book list”-view, you can't select a bunch of items without clicking on each one. This can be very annoying when editing, e.g., metadata for a series of books. Manually selecting 100 books would be incredibly tedious.

Describe the solution you'd like

- I would like to have the “standard” keyboard-shortcut for this. Pressing the shift-key while clicking on the first and last book of the group I want to select. E.g. [Audiobookshelf](https://github.com/advplyr/audiobookshelf) does it this way, which is a very comparable use case.

**Comment by janryklikas-tqgg** [comment]
I was just about to open the exact same issue. 
For me a "select all" button would suffice, but having the possibility to select multiple items in a row with the usual shortcuts would be even better.

For now i use a js snippet in the browser console to select all checkboxes on the page as a workaround, maybe this helps someone else until this feature gets some attention: 

```javascript
document.querySelectorAll('[name="btSelectItem"]').forEach(function(e) {
	e.checked = true;
	e.dispatchEvent(new Event('click'));
});
```

**Comment by SoulOfNoob** [comment]
I was just about to open the exact same issue. 
For me a "select all" button would suffice, but having the possibility to select multiple items in a row with the usual shortcuts would be even better.

For now i use a js snippet in the browser console to select all checkboxes on the page as a workaround, maybe this helps someone else until this feature gets some attention: 

```javascript
document.querySelectorAll('[name="btSelectItem"]').forEach(function(e) {
	e.checked = true;
	e.dispatchEvent(new Event('click'));
});
```

**Comment by lopiv2** [comment]
Indeed, I would go farther. I would add a button to allow selected books to be part of a Selected Shelf. That way we dont have to go one by one adding the book to the wanted Shelf. 
Anyway, keep working on this marvelous app! Great Job!

**Comment by tnaseem** [comment]
Thanks @bosam. I have taken your snippet and added it to a [TamperMonkey](https://www.tampermonkey.net) script, which adds a 'Select All' button to the end of all the others - Clicking it will run your snippet :)

If anyone's interested, this is the code:
```javascript
// ==UserScript==
// @name         Select All
// @namespace    http://tampermonkey.net/
// @version      2025-08-03
// @description  Add a select all button!
// @author       Tarique Naseem
// @match        https://<your-cwa-url>/table*
// @grant        none
// ==/UserScript==

(function() {
    'use strict';

    let btnSelectAll = document.createElement("div");
    btnSelectAll.className = "btn btn-default";
    btnSelectAll.innerHTML = "SELECT ALL";

    btnSelectAll.onclick = () => {
        $('#books-table').bootstrapTable('checkAll');
    };

    let div = document.querySelector("div.row.form-group");
    div.append(btnSelectAll);
})();
```

Replace `<your-cwa-url>` with your Calibre Web URL. In my case it's my reverse proxy address.

Result:

<img width="3238" height="608" alt="Image" src="https://github.com/user-attachments/assets/281f951e-db6a-48db-89ea-dc43da6d4cb6" />

Hope this helps someone!

**Comment by ljo123** [comment]
> Thanks [@bosam](https://github.com/bosam). I have taken your snippet and added it to a [TamperMonkey](https://www.tampermonkey.net) script, which adds a 'Select All' button to the end of all the others - Clicking it will run your snippet :)
> 
> If anyone's interested, this is the code:

This is great thanks! I modified your code a bit to add a "clear all" button too and it works perfectly.
